### PR TITLE
Replaced the deprecated package `k8s.io/utils/pointer` with kubevirt package in `test` and `tools` directory.

### PR DESCRIPTION
**Pull request by anishbista60**
Replaced the deprecated package `k8s.io/utils/pointer` with kubevirt package `kubevirt.io/kubevirt/pkg/pointer` in the  `test` and `tools` directory.

### What this PR does
Before this PR:
- In the  `test` and `tools` directory, we used `k8s.io/utils/pointer` package which is deprecated. 

After this PR:
- Now, it is replaced with kubevirt package `kubevirt.io/kubevirt/pkg/pointer`

Ref #12949

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
/lgtm
Thanks

**Comment by fossedihelm** [comment]
/lgtm cancel
@anishbista60 there are some leftovers

**Comment by anishbista60** [comment]
@fossedihelm  Some new changes were not there in my branch . Now, it's done.

**Comment by anishbista60** [comment]
@alicefr Thank you . Now, we are near to end . Still , two directories are leftover

File: tests/storage/snapshot.go [hunk 1]
```diff
@@ -1397,8 +1396,8 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
 					Expect(found).To(BeTrue())
 				}
 			},
-				Entry("[test_id:4611] without DV garbage collection", pointer.Int32(-1)),
-				Entry("[test_id:8668] with DV garbage collection", pointer.Int32(0)),
+				Entry("[test_id:4611] without DV garbage collection", virtpointer.P(-1)),
+				Entry("[test_id:8668] with DV garbage collection", virtpointer.P(0)),
```
Comment (fossedihelm) [inline tests/storage/snapshot.go:1400]:
```suggestion
				Entry("[test_id:4611] without DV garbage collection", virtpointer.P(int32(-1))),
				Entry("[test_id:8668] with DV garbage collection", virtpointer.P(int32(0))),
```
/hold

**Comment by fossedihelm** [comment]
@anishbista60 Can you address the failure? Thanks

File: tests/storage/snapshot.go (hunk 1 above)
Comment (anishbista60) [reply tests/storage/snapshot.go]: @fossedihelm thank you . Done

**Review by fossedihelm — APPROVED** [review]

**Pull request merged**